src/rust/core/proxy/src/admin.rs

```diff
@@ -0,0 +1,801 @@
+// Copyright 2021 Twitter, Inc.
+// Licensed under the Apache License, Version 2.0
+// http://www.apache.org/licenses/LICENSE-2.0
+
+//! The admin thread, which handles admin requests to return stats, get version
+//! info, etc.
+
+use crate::event_loop::EventLoop;
+use crate::poll::{Poll, LISTENER_TOKEN, WAKER_TOKEN};
+use crate::QUEUE_RETRIES;
+use crate::TCP_ACCEPT_EX;
+use crate::*;
+use common::signal::Signal;
+use common::ssl::{HandshakeError, MidHandshakeSslStream, Ssl, SslContext, SslStream};
+use config::*;
+use core::time::Duration;
+use crossbeam_channel::Receiver;
+use logger::Drain;
+use metrics::{static_metrics, Counter, Gauge, Heatmap};
+use mio::event::Event;
+use mio::{Events, Token, Waker};
+use protocol_admin::*;
+use queues::Queues;
+use session::{Session, TcpStream};
+use std::io::{BufRead, Error, ErrorKind, Write};
+use std::net::SocketAddr;
+use std::sync::Arc;
+use tiny_http::{Method, Request, Response};
+
+static_metrics! {
```

Comment:
Is it time to refactor these common admin features into a util module?

Reply:
Yes. Once merged this can be addressed in a separate PR.

Reply:
#423